cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -2586,8 +2624,17 @@ public TopicsChangedListener getTopicsChangedListener() {
 
     @Override
     public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-        if (requestCode == locationPermissionRequestCode && (geofenceList == null || geofenceList.size() == 0)) {
-            this.initGeoFences();
+        switch (requestCode) {
+            case (pushNotificationPermissionRequestCode):
+                if ((grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) || this.ignoreDisabledNotificationPermission) {
```

Comment:
Please use `this.hasNotificationPermission()` instead of `(grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED)` (please test if it works!)

Result:

```
if (this.hasNotificationPermission() && !this.ignoreDisabledNotificationPermission) {
```